File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.cpp
int first(vector<int>& nums,int target)
{
    int low = 0;
    int high = nums.size()-1;
    int mid;
    int pos = -1;
    while(low<=high)
    { 
        mid= low+(high-low)/2;
        if(nums[mid]>=target)
        {
           high = mid-1;
           if(nums[mid] == target)
           {
                 pos = mid;
           }
            
        }
        else
        {
           low = mid+1;
        } 
    }
     return pos;
}

int second(vector<int>& nums,int target)
{
    int low = 0;
    int high = nums.size()-1;
    int mid;
    int pos = -1;
    while(low<=high)
    { 
        mid= low+(high-low)/2;
        if(nums[mid]<=target)
        {
           low = mid+1;
           if(nums[mid] == target)
           {
             pos = mid;
           }
           
        }
        else
        {
           high = mid-1;
        }
    }
    return pos;
}

class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int count1 = first(nums,target);
        if(count1 == -1)
        {
            return {-1,-1};
        }
        int count2 = second(nums,target);
        return {count1,count2};
    }
};

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int first(int[] nums,int target)
    {
        int low = 0;
        int high = nums.length-1;
        int mid;
        int ans = -1;
        while(low<=high)
        {
            mid = low+(high-low)/2;
            if(nums[mid] >= target)
            {
                high = mid-1;
                if(nums[mid] == target)
                {
                    ans = mid;
                }
            }
            else
            {
                low = mid+1;
            }
        }
        return ans;
    }
       public int second(int[] nums,int target)
    {
        int low = 0;
        int high = nums.length-1;
        int mid;
        int ans = -1;
        while(low<=high)
        {
            mid = low+(high-low)/2;
            if(nums[mid] <= target)
            {
                low = mid+1;
                if(nums[mid] == target)
                {
                    ans = mid;
                }
            }
            else
            {
                high = mid-1;
            }
        }
        return ans;
    }
    public int[] searchRange(int[] nums, int target) {
        int fir = first(nums,target);
        if(fir == -1)
        {
            return new int[]{-1,-1}; 
        }
        int sec = second(nums,target);
        return new int[]{fir,sec};  
    }
}